publictask/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/htmlrenderer/service/HtmlRenderService.kt

```diff
@@ -0,0 +1,29 @@
+package com.ritense.valtimo.backend.plugin.htmlrenderer.service
+
+import com.ritense.valtimo.backend.plugin.htmlrenderer.config.FreemarkerConfig
+import freemarker.template.Template
+import java.io.StringWriter
+
+class HtmlRenderService(
+    private val freemarkerConfig: FreemarkerConfig
+) {
+
+   fun generatePublicTaskHtml(
+       fileName: String,
+       variables: Map<String, Any> = emptyMap()): String =
+    with(StringWriter()) {
+        Template(
+            fileName,
+            getResourceFileAsString("$PUBLIC_TASK_TEMPLATE_PATH/$fileName.ftl"),
+            freemarkerConfig
+        ).process(variables, this)
+        return this.toString()
+    }
+
+    private fun getResourceFileAsString(filePath: String): String? =
+        javaClass.classLoader.getResourceAsStream(filePath)?.bufferedReader().use { it?.readText() }
+
+    companion object {
+        private const val PUBLIC_TASK_TEMPLATE_PATH = "config/template"
```

Comment:
Right now the `/public_task_html.ftl` is located in the `app` module. Which means: It won't be published, people that use this plugin wont' have acces to it and will have to create their own template. Is that intentional?

Reply:
I assume sth changed for the repo by now? Since i am used to that people will just copy the code :) 

The template is a simple example for testing purposes. I will add it to the plugin folder with some comments